paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java

```diff
@@ -230,29 +230,38 @@ public int expireUntil(long earliestId, long endExclusiveId) {
             return 0;
         }
 
-        Set<String> skippingSet = new HashSet<>();
+        Set<String> skippingSet = null;
         try {
-            skippingSet.addAll(snapshotDeletion.manifestSkippingSet(skippingSnapshots));
+            skippingSet = new HashSet<>(snapshotDeletion.manifestSkippingSet(skippingSnapshots));
         } catch (Exception e) {
-            // maybe snapshot been deleted by other jobs.
-            if (e.getCause() == null || !(e.getCause() instanceof FileNotFoundException)) {
```

Comment:
Add tests for this bug.

Reply:
ExpireSnapshotsTest#testManifestFileSkippingSetFileNotFoundException

It will fail on master.